### PR TITLE
Skip NameObject when building outline

### LINKED ISSUE #193: PyPDF2.utils.PdfReadError: Unexpected destination '/__WKANCHOR_2'
- PyPDF2 1.24

The exception occurs when building outline for a pdf generated by wkhtmltopdf. 

```python
from PyPDF2 import PdfFileMerger, PdfFileReader

merger = PdfFileMerger()
merger.append(PdfFileReader(file('test.pdf', 'rb')))  # Unexpected destination

_pdf = PdfFileReader(file('a.pdf', 'rb'))
_pdf.outlines  # same as the exception
```

I got a below error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kay/Workspace/.virtualenv/rsquare-doc/lib/python2.7/site-packages/PyPDF2/pdf.py", line 1145, in <lambda>
    outlines = property(lambda self: self.getOutlines(), None, None)
  File "/Users/kay/Workspace/.virtualenv/rsquare-doc/lib/python2.7/site-packages/PyPDF2/pdf.py", line 1180, in getOutlines
    outline = self._buildOutline(node)
  File "/Users/kay/Workspace/.virtualenv/rsquare-doc/lib/python2.7/site-packages/PyPDF2/pdf.py", line 1224, in _buildOutline
    raise utils.PdfReadError("Unexpected destination %r" % dest)
PyPDF2.utils.PdfReadError: Unexpected destination '/__WKANCHOR_2'
```

Any help on this would be appreciated. Thank you.

### PR DESCRIPTION
**Pull request by LamerLink**
Fixes #193, uses @hannal's solution to resolve read/merge issues with wkhtmltopdf pdfs by skipping NameObject entities. Related to #778.

### CODE REVIEW

**Comment by LamerLink** [comment]
@MartinThoma Think I got it now. Thanks again for the help.

**Comment by LamerLink** [comment]
Never mind, need to update some tests it seems.

**Comment by LamerLink** [comment]
I tried numerous "bad" PDFs but couldn't get this error to raise. Will have to look into creating a PDF with garbage anchors and uploading to the samples repo.

**Comment by MartinThoma** [comment]
@LamerLink Could it be that your PR just solved an issue that `test_unexpected_destination` has? Maybe we could simply adjust the test to:

```python
def test_unexpected_destination():
    url = "https://corpora.tika.apache.org/base/docs/govdocs1/913/913678.pdf"
    name = "tika-913678.pdf"
    reader = PdfReader(BytesIO(get_pdf_from_url(url, name=name)))
    merger = PdfMerger()
    merger.append(reader)
```

in this PR?

**Comment by LamerLink** [comment]
@MartinThoma Good call, updated. I thought about adding an `assert len(reader.outlines) == 0` since there'll be no valid destinations added to the outline in this case, but I decided not too since it's a little ubiquitous fir this case. Not to mention some normal PDFs can have no outlines, especially if they're pure images. 

Looks like all checks have passed!

**Comment by MartinThoma** [comment]
@LamerLink Thank you for your effort. There was another PR #1076 which also solved the same issue, but that one is clearer to me why it solves the issue. As the issue is now (hopefully) gone, this PR seems no longer necessary.

Please let me know if I got something wrong - I can always re-open a PR

**Comment by LamerLink** [comment]
@MartinThoma No problem at all, I read over that PR and it makes sense to me! Thanks for all the effort.

**Comment by MartinThoma** [comment]
I've just noticed that your pr increases coverage quite a bit. That is unexpected to me. I want to understand that better

File: tests/test_reader.py [hunk 1]
```diff
@@ -812,6 +812,14 @@ def test_get_fields_read_write_report():
     os.remove("tmp-fields-report.txt")
 
 
+def test_unexpected_destination():
```
Comment (MartinThoma) [inline tests/test_reader.py:815]:
```suggestion
def test_unexpected_destination():
    from PyPDF2 import PdfMerger
```

**Comment by MartinThoma** [comment]
I'm not sure why it originally showed that the test coverage was increased by that much. If we would do it now with https://github.com/py-pdf/PyPDF2/pull/1158, the coverage would decrease slightly (which is what I expected)

**Pull request closed without merging**